files/blockinfile.py

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2014, 2015 YAEGASHI Takeshi <yaegashi@debian.org>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import re
+import os
+import tempfile
+
+DOCUMENTATION = """
+---
+module: blockinfile
+author:
+    - 'YAEGASHI Takeshi (@yaegashi)'
+extends_documentation_fragment:
+    - files
+    - validate
+short_description: Insert/update/remove a text block
+                   surrounded by marker lines.
+version_added: '2.0'
+description:
+  - 'This module will insert/update/remove a block of multi-line text
+    surrounded by customizable marker lines
+    (default: "# {BEGIN/END} ANSIBLE MANAGED BLOCK").'
+options:
+  dest:
+    aliases: [ name, destfile ]
+    required: true
+    description:
+      - The file to modify.
+  marker:
+    required: false
+    default: '# {mark} ANSIBLE MANAGED BLOCK'
+    description:
+      - The marker line template.
+        "{mark}" will be replaced with "BEGIN" or "END".
+  block:
+    aliases: [ content ]
+    required: false
+    default: ''
+    description:
+      - The text to insert inside the marker lines.
+        If it's an empty string, the marker lines will also be removed.
+  insertafter:
+    required: false
+    default: EOF
+    description:
+      - If specified, the block will be inserted after the last match of
+        specified regular expression. A special value is available; C(EOF) for
```

Comment:
It is [suggested](http://docs.ansible.com/ansible/developing_modules.html#example) to use C() for file and option names, and I() when referencing parameters. Here the `C(EOF)` should be used.